vcnc-rest/lib/rethink.js

```diff
@@ -0,0 +1,89 @@
+/*
+ *    Copyright (C) 2017    IC Manage Inc.
+ *
+ *    See the file 'COPYING' for license information.
+ */
+'use strict'; // eslint-disable-line strict
+const config = require('./configuration.js');
+const r = require('rethinkdb');
+const dbName = config.rethinkdb.connection.db;
+const tableName = 'about';
+let cnxtn = null;
+
+const schemaMajor = 1;
+const schemaMinor = 0;
+
+function newAboutRow() {
+  return r.table(tableName).insert({
+    schemaMajor,
+    schemaMinor,
+    timestamp: r.now()
+  }).run(cnxtn);
+}
+
+/**
+ *  Initializes the rethinkdb module, creating a connection object
+ *  and creating the database if necessary.
+ *
+ *  Must be called before any other rethinkdb related module.
+ *
+ *  @return {promise} A promise fulfilled when the connection is ready.
+ */
+function init() {
+  if (!config.test.enable_grid_jobs) return Promise.resolve();
+  return r.connect(config.rethinkdb.connection)
+  .then(conn => {
+    cnxtn = conn;  // save the connection for later reuse
+    return r.dbList().filter(v => v.eq(dbName)).run(conn);
+  })
+  .then(lst => {
+    //  Create the database if necessary.
+    if (lst.length === 0) {
+      return r.dbCreate(dbName).run(cnxtn);
+    }
+    return Promise.resolve();
+  })
+  .then(() =>
+    //  Look for the table in the database
+    r.tableList().filter(v => v.eq(tableName)).run(cnxtn)
+  )
+  .then(lst => {
+    // Create the table if necessary.
+    if (lst.length === 0) {
+      return r.tableCreate(tableName).run(cnxtn);
+    }
+    return Promise.resolve();
+  })
+  .then(() => r.table(tableName).orderBy(r.desc('timestamp')).limit(1).run(cnxtn))
+  .then(lst => {
```

Comment:
Originally, that's what I intended to do.  But then I realized it might be useful to store the upgrade history of the database.  For example, suppose we discovered there was a bug in the conversion software from schemaMajor 3 to 4, or that the upgrade from 5.2 to 5.3 wasn't actually backward compatible.  The history would give us a way to identify databases which might be corrupted and need to be fixed (or abandonded).